algorithms/kernel/kernel_function/kernel_function_linear_dense_default_impl.i

```diff
@@ -134,21 +134,29 @@ services::Status KernelImplLinear<defaultDense, algorithmFPType, cpu>::computeIn
 
     if (a1 != a2)
     {
-        ReadRows<algorithmFPType, cpu> mtA1(*const_cast<NumericTable *>(a1), 0, nVectors1);
-        DAAL_CHECK_BLOCK_STATUS(mtA1);
-        const algorithmFPType * dataA1 = const_cast<algorithmFPType *>(mtA1.get());
-
         ReadRows<algorithmFPType, cpu> mtA2(*const_cast<NumericTable *>(a2), 0, nVectors2);
         DAAL_CHECK_BLOCK_STATUS(mtA2);
         const algorithmFPType * dataA2 = const_cast<algorithmFPType *>(mtA2.get());
 
-        WriteOnlyRows<algorithmFPType, cpu> mtR(r, 0, nVectors1);
-        DAAL_CHECK_BLOCK_STATUS(mtR);
-        algorithmFPType * dataR = mtR.get();
+        const size_t blockSize = 256;
+        const size_t blockNum  = nVectors1 / blockSize + !!(nVectors1 % blockSize);
+
+        daal::threader_for(blockNum, blockNum, [&](const size_t iBlock) {
```

Comment:
Do you see perf difference with xgemm_blocked method?

Reply:
threader_for+gemm performs a little bit better (~5%) than xgemm_blocked. So I think I shouldn't use xgemm_blocked at all

Reply:
xgemm_blocked - it's the same as threader_for + gemm :)

Reply:
Agree with Egor's comment about xgemm_blocked. Also, you can pass block sizes for matrices a1 and a2 into xgemm_blocked as parameters. And 5% performance difference can be a deviation in measurements.